pipeline/subtraction.py

```diff
@@ -225,49 +236,335 @@ def _subtract_zogy(self,
         output['score'] = output.pop('score_corr')
         output['alpha_err'] = output.pop('alpha_std')
+        # convert flux based into magnitude based zero point
+        output['zero_point'] = 2.5 * np.log10(output['zero_point'])
 
         outwt, outfl = zogy_add_weights_flags(
             ref_image.weight,
             new_image.weight,
             ref_image.flags,
             new_image.flags,
+            ref_zp.zp,
+            new_zp.zp,
+            output['zero_point'],
             ref_psf.fwhm_pixels,
             new_psf.fwhm_pixels
         )
         output['outwt'] = outwt
         output['outfl'] = outfl
 
-        # convert flux based into magnitude based zero point
-        output['zero_point'] = 2.5 * np.log10(output['zero_point'])
 
         return output
 
-    def _subtract_hotpants(self, new_image, ref_image):
-        """Use Hotpants to subtract the two images.
+    def _subtract_hotpants(self,
+                           new_image, new_bg, new_sources, new_wcs, new_psf,
+                           ref_image, ref_bg, ref_sources, ref_wcs, ref_psf ):
+        """Alard/Lupton subtraction.
+
+        This applies PSF matching and uses the Alard/Lupton algorithm,
+        as implemented by "hotpants", to subtract the two images.
 
-        This applies PSF matching and uses the Hotpants algorithm to subtract the two images.
-        reference: ...
+           https://github.com/acbecker/hotpants
+           https://ui.adsabs.harvard.edu/abs/1998ApJ...503..325A/abstract
 
         Parameters
         ----------
         new_image : Image
-            The Image containing the new data, including the data array, weight, and flags.
-            Image must also have the PSF and ZeroPoint objects loaded.
+            The Image containing the new data, including the data array,
+            weight, and flags.  ref_image and new_image must be aligned
+            to each other.
+
+        new_bg : Background
+            Sky background for new_image
+
+        new_sources : SourceList
+            Source List for new_image.  Must have the is_star parameter
+            (which, as of this writing, means that the source list must
+            be a sextrfits source list.)
+
+        new_wcs : WorldCoordinates
+            WCS  for new_image
+
+        new_psf : PSF
+            PSF for new_image
+
         ref_image : Image
-            The Image containing the reference data, including the data array, weight, and flags
-            Image must also have the PSF and ZeroPoint objects loaded.
+            The Image containing the reference data, including the data
+            array, weight, and flags.  ref_image and new_image must be
+            aligned to each other.
+
+        ref_sources: SourceList
+            Source List for ref_image.  Must have the is_star parameter
+            (which, as of this writing, means that the source list must
+            be a sextrfits source list.)
```

Comment:
As a fleeting thought, do you think it would be better to say something like "as of Nov 2024" to add context the next time we come around to this? This might be be too minor to worry about.

Reply:
Yeah, maybe.  I often put that in as a disclaimer because *all the time* things change somewhere else in the code, and we don't think to update the documentation in other places.  (At least one change in this PR was me fixing documentation for something that changed several PRs ago.)